integration/disk_snapshot_test.go

```diff
@@ -0,0 +1,103 @@
+// +build integration
+
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/integration/generate"
+	"github.com/m3db/m3db/persist/fs"
+	xtime "github.com/m3db/m3x/time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDiskSnapshotSimple(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow() // Just skip if we're doing a short run
+	}
+	// Test setup
+	testOpts := newTestOptions(t).
+		SetTickMinimumInterval(time.Second)
+	testSetup, err := newTestSetup(t, testOpts, nil)
+	require.NoError(t, err)
+	defer testSetup.close()
+
+	md := testSetup.namespaceMetadataOrFail(testNamespaces[0])
+	blockSize := md.Options().RetentionOptions().BlockSize()
+	filePathPrefix := testSetup.storageOpts.CommitLogOptions().FilesystemOptions().FilePathPrefix()
+
+	// Start the server
+	log := testSetup.storageOpts.InstrumentOptions().Logger()
+	log.Debug("disk flush test")
+	require.NoError(t, testSetup.startServer())
+	log.Debug("server is now up")
+
+	// Stop the server
+	defer func() {
+		require.NoError(t, testSetup.stopServer())
+		log.Debug("server is now down")
+	}()
+
+	// Write test data
+	now := testSetup.getNowFn()
+	seriesMaps := make(map[xtime.UnixNano]generate.SeriesBlock)
+	inputData := []generate.BlockConfig{
```

Comment:
maybe write out data for multiple namespaces too?

Reply:
I am:

```golang
for _, ns := range testSetup.namespaces {
    require.NoError(t, testSetup.writeBatch(ns.ID(), testData))
}
```

Reply:
fair enough.